src/cider/tests/benchmark/ArrowDataReader.h

```diff
@@ -30,55 +30,70 @@
-  auto tag = chunk_arrays->type()->id();
+substrait::Type convertToSubstraitType(const arrow::Type::type tag) {
   switch (tag) {
+    case arrow::Type::type::BOOL:
+      return CREATE_SUBSTRAIT_TYPE(Bool);
+    case arrow::Type::type::INT8:
+      return CREATE_SUBSTRAIT_TYPE(I8);
+    case arrow::Type::type::INT16:
+      return CREATE_SUBSTRAIT_TYPE(I16);
     case arrow::Type::type::INT32:
-      constructArrowFixedColumn(CREATE_SUBSTRAIT_TYPE(I32));
+      return CREATE_SUBSTRAIT_TYPE(I32);
     case arrow::Type::type::INT64:
-      constructArrowFixedColumn(CREATE_SUBSTRAIT_TYPE(I64));
+      return CREATE_SUBSTRAIT_TYPE(I64);
     case arrow::Type::type::FLOAT:
-      constructArrowFixedColumn(CREATE_SUBSTRAIT_TYPE(Fp32));
+      return CREATE_SUBSTRAIT_TYPE(Fp32);
     case arrow::Type::type::DOUBLE:
-      constructArrowFixedColumn(CREATE_SUBSTRAIT_TYPE(Fp64));
+      return CREATE_SUBSTRAIT_TYPE(Fp64);
     case arrow::Type::type::STRING:
-      constructArrowVariColumn(CREATE_SUBSTRAIT_TYPE(Varchar));
+      return CREATE_SUBSTRAIT_TYPE(Varchar);
+    case arrow::Type::type::DATE32:
+      return CREATE_SUBSTRAIT_TYPE(Date);
+    case arrow::Type::type::TIME64:
+      return CREATE_SUBSTRAIT_TYPE(Time);
+    case arrow::Type::type::TIMESTAMP:
+      return CREATE_SUBSTRAIT_TYPE(Timestamp);
+    default:
+      CIDER_THROW(CiderRuntimeException, "Failed to convert arrow type: " + tag);
+  }
+}
+
+void constructArrowArray(ArrowArrayBuilder& builder,
+                         std::shared_ptr<arrow::ChunkedArray> chunk_arrays,
+                         const std::string& name) {
+  for (int i = 0; i < chunk_arrays->num_chunks(); i++) {
+    auto arrow_array = chunk_arrays->chunk(i);
+    auto tag = arrow_array->type_id();
+    int64_t buffer_num = arrow_array->data()->buffers.size();
+    switch (buffer_num) {
+      case 2:
+        builder.addColumn(name,
+                          convertToSubstraitType(tag),
+                          arrow_array->null_bitmap_data(),
+                          arrow_array->data()->buffers[1]->data());
+        break;
+      case 3:
+        builder.addColumn(name,
+                          convertToSubstraitType(tag),
+                          arrow_array->null_bitmap_data(),
+                          arrow_array->data()->buffers[1]->data(),
+                          arrow_array->data()->buffers[2]->data());
+        break;
+      default:
+        CIDER_THROW(CiderRuntimeException,
+                    "Failed to extract array buffer, buffe_num is : " + buffer_num);
+    }
   }
 }
+
 }  // namespace
 
 class ArrowDataReader {
  public:
-  ArrowDataReader(std::string file_name, std::vector<std::string>& col_name)
-      : file_name_{file_name}, col_name_{col_name} {};
-
+  ArrowDataReader(const std::string& file_name,
+                  const std::unordered_set<std::string>& skip_cols = {})
+      : file_name_{file_name}, skip_cols_{skip_cols} {};
```

Comment:
I think you can provide 2 API: `ArrowDataReader(const std::string& file_name)` and `ArrowDataReader(const std::string& file_name, std::vector<std::string> col_names)`. The first one will read all cols and the second one will read requested cols with default order as occur in original table. Seems `skip_cols` is hard to understand and not easy if we want read 2 columns out of a 20 cols table.